src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs

```diff
@@ -1452,6 +1452,7 @@ public async Task WaitForExitAsync(CancellationToken cancellationToken = default
                 // exception up to the user
                 if (HasExited)
                 {
+                    await WaitUntilOutputEOF().ConfigureAwait(false);
```

Comment:
Is it possible this might throw?  If so, is that the exception we want to bubble up rather than the one being handled here?

Reply:
> Is it possible this might throw?

It seems like it can.

> If so, is that the exception we want to bubble up rather than the one being handled here?

I think so, the comments seem to indicate that the handled exception should not be surfaced if the process has exited. At the same time, it makes sense to bubble up any exceptions thrown by the wait method.